lsp/nls/src/background.rs

```diff
@@ -0,0 +1,347 @@
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    path::PathBuf,
+    process::Child,
+    time::{Duration, Instant},
+};
+
+use crossbeam::channel::{Receiver, Select, Sender};
+use ipc_channel::ipc::{IpcOneShotServer, IpcReceiver, IpcSender};
+use log::warn;
+use lsp_types::Url;
+use nickel_lang_core::{
+    cache::SourcePath,
+    eval::{cache::CacheImpl, VirtualMachine},
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{diagnostic::SerializableDiagnostic, files::uri_to_path, world::World};
+
+const EVAL_TIMEOUT: Duration = Duration::from_secs(1);
```

Comment:
I guess experience will tell, but I think one second might be a bit harsh

Reply:
That's quite possible, and I need to do some more experimentation with big projects. My reason for being conservative here is that we re-evaluate from scratch pretty often (i.e. every time the file contents get updated) and I don't want to be constantly pegging the cpu.